### PR TITLE
Fix codegen for StoreNonTemporal

### LINKED ISSUE #23509: Test failure: JIT_HardwareIntrinsics._X86_Regression_GitHub_23438_GitHub_23438_r_GitHub_23438_r_/_X86_Regression_GitHub_23438_GitHub_23438_r_GitHub_23438_r_sh
Opened on behalf of @AriNuer

The test `JIT_HardwareIntrinsics._X86_Regression_GitHub_23438_GitHub_23438_r_GitHub_23438_r_/_X86_Regression_GitHub_23438_GitHub_23438_r_GitHub_23438_r_sh` has failed.

Failure Message:
```

Unhandled Exception: System.NullReferenceException: Object reference not set to an instance of an object.
   at GitHub_23438.Program.TestSse2X64StoreNonTemporal_Int64() in /Users/vsts/agent/2.148.2/work/1/s/tests/src/JIT/HardwareIntrinsics/X86/Regression/GitHub_23438/GitHub_23438.cs:line 145
   at GitHub_23438.Program.Main(String[] args) in /Users/vsts/agent/2.148.2/work/1/s/tests/src/JIT/HardwareIntrinsics/X86/Regression/GitHub_23438/GitHub_23438.cs:line 33
/Users/dotnet-bot/dotnetbuild/work/347da326-d322-4618-8fb0-3a49bdc72fb5/Work/c19817da-af16-4d4a-919a-7054de84940a/Exec/X86/Regression/GitHub_23438/GitHub_23438_r/GitHub_23438_r.sh: line 252: 49625 Abort trap: 6           $LAUNCHER $ExePath \"${CLRTestExecutionArguments[@]}\"
/n/nReturn code:      1/nRaw output file:      /Users/dotnet-bot/dotnetbuild/work/347da326-d322-4618-8fb0-3a49bdc72fb5/Work/c19817da-af16-4d4a-919a-7054de84940a/Exec/Reports/JIT.HardwareIntrinsics/X86/Regression/GitHub_23438/GitHub_23438_r/GitHub_23438_r.output.txt/nRaw output:/nBEGIN EXECUTION
/Users/dotnet-bot/dotnetbuild/work/347da326-d322-4618-8fb0-3a49bdc72fb5/Payload/corerun GitHub_23438_r.exe ''
Expected: 100
Actual: 134
END EXECUTION - FAILED
Test Harness Exitcode is : 1
/nTo run the test:/n> set CORE_ROOT=/Users/dotnet-bot/dotnetbuild/work/347da326-d322-4618-8fb0-3a49bdc72fb5/Payload/n> /Users/dotnet-bot/dotnetbuild/work/347da326-d322-4618-8fb0-3a49bdc72fb5/Work/c19817da-af16-4d4a-919a-7054de84940a/Exec/X86/Regression/GitHub_23438/GitHub_23438_r/GitHub_23438_r.sh/n
Expected: True
Actual:   False
```

Stack Trace:
```
   at JIT_HardwareIntrinsics._X86_Regression_GitHub_23438_GitHub_23438_r_GitHub_23438_r_._X86_Regression_GitHub_23438_GitHub_23438_r_GitHub_23438_r_sh() in /Users/vsts/agent/2.148.2/work/1/s/bin/tests/OSX.x64.Release/TestWrappers/JIT.HardwareIntrinsics/JIT.HardwareIntrinsics.XUnitWrapper.cs:line 9631
```

Build : 3.0 - 20190327.73 (Core Tests)
Failing configurations:
- OSX.1014.Amd64-x64
  - 
  - 
- OSX.1013.Amd64-x64
  - 
  - 
- OSX.1012.Amd64-x64
  - 
  - 

Details:
https://mc.dot.net/#/product/netcore/30/source/official~2Fdotnet~2Fcoreclr~2Frefs~2Fheads~2Fmaster/type/test~2Ffunctional~2Fcli~2F/build/20190327.73/workItem/JIT.HardwareIntrinsics/analysis/xunit/JIT_HardwareIntrinsics._X86_Regression_GitHub_23438_GitHub_23438_r_GitHub_23438_r_~2F_X86_Regression_GitHub_23438_GitHub_23438_r_GitHub_23438_r_sh

### PR DESCRIPTION
**Pull request by CarolEidt**
Also, add some asserts and mark some intrinsics as not supporting containment.

Fix #23509

### CODE REVIEW

File: src/jit/hwintrinsiccodegenxarch.cpp [hunk 1]
```diff
@@ -1680,10 +1684,11 @@ void CodeGen::genSSE2Intrinsic(GenTreeHWIntrinsic* node)
...
             assert(op2 != nullptr);
 
-            op2Reg          = op2->gtRegNum;
-            instruction ins = HWIntrinsicInfo::lookupIns(intrinsicId, baseType);
-            op1Reg          = op1->gtRegNum;
-            emit->emitIns_AR_R(ins, emitTypeSize(baseType), op2Reg, op1Reg, 0);
+            op2Reg                = op2->gtRegNum;
+            instruction ins       = HWIntrinsicInfo::lookupIns(intrinsicId, baseType);
+            op1Reg                = op1->gtRegNum;
+            GenTreeStoreInd store = storeIndirForm(node->TypeGet(), op1, op2);
```
Comment (AndyAyersMS) [inline src/jit/hwintrinsiccodegenxarch.cpp]: You no longer need to set `op1Reg` or `op2Reg` here ?

Comment (CarolEidt) [reply src/jit/hwintrinsiccodegenxarch.cpp]: Good catch; fixed.

File: src/jit/hwintrinsiclistxarch.h [hunk 2]
```diff
@@ -497,7 +497,7 @@ HARDWARE_INTRINSIC(AVX2_HorizontalSubtractSaturate,                 "HorizontalS
...
 HARDWARE_INTRINSIC(AVX2_LoadAlignedVector256NonTemporal,            "LoadAlignedVector256NonTemporal",              AVX2,         -1,              32,           1,     {INS_movntdqa,          INS_movntdqa,       INS_movntdqa,       INS_movntdqa,       INS_movntdqa,       INS_movntdqa,       INS_movntdqa,       INS_movntdqa,       INS_invalid,        INS_invalid},           HW_Category_MemoryLoad,             HW_Flag_NoFlag)
 HARDWARE_INTRINSIC(AVX2_MaskLoad,                                   "MaskLoad",                                     AVX2,         -1,               0,           2,     {INS_invalid,           INS_invalid,        INS_invalid,        INS_invalid,        INS_vpmaskmovd,     INS_vpmaskmovd,     INS_vpmaskmovq,     INS_vpmaskmovq,     INS_invalid,        INS_invalid},           HW_Category_MemoryLoad,             HW_Flag_UnfixedSIMDSize)
```
Comment (tannergooding) [inline src/jit/hwintrinsiclistxarch.h]: Does `MaskLoad` also need to be marked `NoContainment` for now?

File: src/jit/hwintrinsiccodegenxarch.cpp [hunk 3]
```diff
@@ -2128,6 +2131,8 @@ void CodeGen::genBMI1OrBMI2Intrinsic(GenTreeHWIntrinsic* node)
         case NI_BMI2_MultiplyNoFlags:
         case NI_BMI2_X64_MultiplyNoFlags:
         {
+            // These do not support containment
+            assert(!op2->isContained());
```
Comment (tannergooding) [inline src/jit/hwintrinsiccodegenxarch.cpp]: Many of these are just temporary, correct? That is we should be able to make them containable eventually and change `emitIns_AR_R` into `emitInsStoreInd` (in a separate PR)?

File: src/jit/hwintrinsiclistxarch.h (hunk 2 above)
Comment (CarolEidt) [reply src/jit/hwintrinsiclistxarch.h]: No, it's explicitly handled here in `Lowering`: https://github.com/dotnet/coreclr/blob/master/src/jit/lowerxarch.cpp#L2952 and here in `CodeGen`:  https://github.com/dotnet/coreclr/blob/master/src/jit/hwintrinsiccodegenxarch.cpp#L184.

However, as @fiigii pointed out, we need to expand our testing to cover more containment scenarios.

File: src/jit/hwintrinsiccodegenxarch.cpp (hunk 3 above)
Comment (CarolEidt) [reply src/jit/hwintrinsiccodegenxarch.cpp]: Yes, it should be possible, but I didn't want to tackle that complexity at this point.

**Comment by sandreenko** [comment]
@dotnet/dnceng could somebody look at this PR?9 build jobs are shown as pending but that have not started according to [ADO](https://dev.azure.com/dnceng/public/_build/results?buildId=136281):
image
image


There are many free machines in each build pool and there are succesful build tasks there were run later.

**Comment by CarolEidt** [comment]
@dotnet/jit-contrib @dotnet/dnceng - any recommendations for how to get the tests completed? (Or merge without waiting for completion), as this is fixing an existing test failure?

**Comment by MattGal** [comment]
@CarolEidt looking now...

**Comment by MattGal** [comment]
@CarolEidt @mmitche just got back from some training, and I'll talk with him about it.  I can't find evidence of anything going wrong from the telemetry we have, both from the pool provider and Helix.  We'll keep investigating, I'd say if the tests pass in a couple places just merge and if this keeps happening let me know.

**Pull request merged**